### PR TITLE
[Auto para] Relaunch with auto mapping function

### PR DESCRIPTION
**Pull request by aoyulong**
### PR types
New features

### PR changes
Others

### Describe

The pr will relaunch the distributed training based on the rank mapping file produced by the auto mapping function. So there will be two times of launching.

1.  The first launch will produce the rank mapping file based on the distributed programs and the cluster topology.
2. The second launch will use the rank mapping file to map the rank and the corresponding device for the actual distributed training.

### CODE REVIEW

File: python/paddle/distributed/fleet/launch.py [hunk 1]
```diff
@@ -287,20 +280,43 @@ def cpuonly_check(args):
...
-    # lazy launch for auto-parallel
+    # auto mapping between processes and devices for auto-parallel
     if args.enable_auto_mapping == True:
-        cluster, pod = get_mapped_cluster_from_args(args, device_mode)
+        assert args.cluster_topo_path is not None, \
+            "The cluster topology must be provied when enabling auto mapping."
+        if args.rank_mapping_path is None:
+            # original_args = [shlex.quote(c) for c in sys.argv[1:]]
+            original_args = sys.argv[1:]
+            os.environ["PADDLE_ORIGINAL_CMD_ARGS"] = " ".join(original_args)
```
Comment (kuizhiqing) [inline python/paddle/distributed/fleet/launch.py:325]: this part looks fragile

File: python/paddle/distributed/fleet/launch.py [hunk 2]
```diff
@@ -165,25 +166,17 @@ def _parse_args():
...
+        type=str,
+        default=None,
+        help="A json format file will be stored in this path which is used"
+        "to represent the cluster topology information for auto parallel.")
+    collective_group.add_argument(
+        "--rank_mapping_path",
+        type=str,
+        default=None,
+        help="A json format file will be stored in this path which is used"
+        "to map processes to machines for auto parallel.")
```
Comment (kuizhiqing) [inline python/paddle/distributed/fleet/launch.py:188]: Add one more config file is expensive, is it possible to use _xxx_config_ to hold all ? may be  _paddle_config_ and you hold some sections ?

File: python/paddle/distributed/fleet/launch.py (hunk 1 above)
Comment (aoyulong) [reply python/paddle/distributed/fleet/launch.py:325]: > 
This has been dealt with by shlex.split on line 154 of the above parallelizer.py.

File: python/paddle/distributed/fleet/launch.py (hunk 2 above)
Comment (aoyulong) [reply python/paddle/distributed/fleet/launch.py:188]: > 

The rank_mapping file will be automatically generated by our framework in the pre-launch analysis pass and must not be exposed to users.

**Review by kuizhiqing — APPROVED** [review]
LGTM

**Review by XieYunshen — APPROVED** [review]
LGTM for `set_tests_properties(test_auto_parallel_relaunch PROPERTIES LABELS "RUN_TYPE=EXCLUSIVE" TIMEOUT 120)`

**Review by fuyinno4 — APPROVED** [review]

**Review by fuyinno4 — APPROVED** [review]

**Review by JZ-LIANG — APPROVED** [review]
LGTM

**Pull request merged**